### PR TITLE
GROK-11348: CI: Build dockerfiles for packages in GitHub Actions

### PR DESCRIPTION
**Pull request by ksumag**
fix:
- package name and version now gets via steps.outputs
- white spaces has been deleted
- used jq -r instead sed
- add build image condition steps.publish.outcome is skipped or a failure

### CODE REVIEW

File: .github/workflows/packages.yml [hunk 1]
```diff
@@ -230,6 +230,67 @@ jobs:
...
+          files: "dockerfiles"  
+      - name: check if dockerfies directory not empty 
+        id: NEdockerdir
+        run: |
+          if [ -z "$(ls -A  packages/${{ matrix.package }}/dockerfiles)" ]; then
+          echo "::set-output name=DIR_EMPTY::true"
+          else
+          echo "::set-output name=DIR_EMPTY::false"
+          fi
+      - name: Check if directory has changes
```
Comment (dnillovna) [inline .github/workflows/packages.yml:247]:
```suggestion
      - name: Check if the dockerfiles directory has changes
```

File: .github/workflows/packages.yml [hunk 2]
```diff
@@ -230,6 +230,67 @@ jobs:
...
+        if: steps.check_directory.outputs.files_exists == 'true' && steps.NEdockerdir.outputs.DIR_EMPTY == 'false' && steps.publish.outcome == 'success'
+
+      - name: get docker image version
+        id: docker_name_ver
+        working-directory: packages/${{ matrix.package }}
+        run: |
+          echo "::set-output name=IMAGE_VER::$(jq -r .version package.json)"
+          echo "::set-output name=PACKAGE_NAME::$(jq -r .version package.json | sed -e 's,@datagrok/,,g')"
+ #         echo "IMAGE_VER=`echo "$(jq .version package.json | sed -E -e 's/(^"|"$)//g')"`" >> $GITHUB_ENV
+ #         echo "PACKAGE_NAME=`echo "$(jq .name package.json | sed -E -e 's/(^"|"$)//g' -e 's,@datagrok/,,g')"`" >> $GITHUB_ENV
```
Comment (dnillovna) [inline .github/workflows/packages.yml:267]:
```suggestion
```

File: .github/workflows/packages.yml [hunk 3]
```diff
@@ -230,6 +230,67 @@ jobs:
...
+
+      - name: Build and push package dockerimages
```
Comment (dnillovna) [inline .github/workflows/packages.yml:269]:
```suggestion
      - name: Build and push package Docker image
```

File: .github/workflows/packages.yml [hunk 4]
```diff
@@ -230,6 +230,67 @@ jobs:
...
+        if: steps.check_directory.outputs.files_exists == 'true' && steps.NEdockerdir.outputs.DIR_EMPTY == 'false' && steps.publish.outcome == 'success'
+        uses: docker/build-push-action@v3
+        with:
+          context: packages/${{ matrix.package }}/dockerfiles
+          push: true
+          tags: datagrok/${{ steps.docker_name_ver.outputs.PACKAGE_NAME }}:${{ steps.docker_name_ver.outputs.IMAGE_VER }}
+        
+
+
```
Comment (dnillovna) [inline .github/workflows/packages.yml:278]:
```suggestion

```

File: .github/workflows/packages.yml [hunk 5]
```diff
@@ -230,6 +230,67 @@ jobs:
...
+        
+
+    
+      - name: build dockerfiles as artifact when npm push failed or skipped
+        if: steps.directory_changes_checker.output.hasChanges == 'true' && steps.publish.outcome == 'failure' || steps.publish.outcome == 'skipped'
+        uses: docker/build-push-action@v3
+        with:
+          context: packages/${{ matrix.package }}/dockerfiles
+          push: false
+          tags: datagrok/${{ steps.docker_name_ver.outputs.PACKAGE_NAME }}:${{ steps.docker_name_ver.outputs.IMAGE_VER }}${{ steps.datagrok-image.outputs.commit_sha }}
```
Comment (dnillovna) [inline .github/workflows/packages.yml:285]:
```suggestion
          tags: datagrok/${{ steps.docker_name_ver.outputs.PACKAGE_NAME }}:${{ steps.docker_name_ver.outputs.IMAGE_VER }}-${{ steps.datagrok-image.outputs.commit_sha }}
```

File: .github/workflows/packages.yml [hunk 6]
```diff
@@ -230,6 +230,67 @@ jobs:
...
+        
+      - name: Upload dockerfiles as artifact when npm push failed or skipped
+        if: steps.directory_changes_checker.output.hasChanges == 'true' && steps.publish.outcome == 'failure' || steps.publish.outcome == 'skipped'
```
Comment (dnillovna) [inline .github/workflows/packages.yml:288]:
```suggestion
        if: steps.directory_changes_checker.output.hasChanges == 'true' && (steps.publish.outcome == 'failure' || steps.publish.outcome == 'skipped')
```

File: .github/workflows/packages.yml [hunk 7]
```diff
@@ -230,6 +230,67 @@ jobs:
...
+        uses: ishworkh/docker-image-artifact-upload@v1
+        with:
+          image: "datagrok/${{ steps.docker_name_ver.outputs.PACKAGE_NAME }}:${{ steps.docker_name_ver.outputs.IMAGE_VER }}${{ steps.datagrok-image.outputs.commit_sha }}"
```
Comment (dnillovna) [inline .github/workflows/packages.yml:291]:
```suggestion
          image: "datagrok/${{ steps.docker_name_ver.outputs.PACKAGE_NAME }}:${{ steps.docker_name_ver.outputs.IMAGE_VER }}-${{ steps.datagrok-image.outputs.commit_sha }}"
```

File: .github/workflows/packages.yml [hunk 8]
```diff
@@ -230,6 +230,67 @@ jobs:
...
+        
+
```
Comment (dnillovna) [inline .github/workflows/packages.yml:293]:
```suggestion
```

File: .github/workflows/packages.yml [hunk 9]
```diff
@@ -405,4 +466,4 @@ jobs:
               fi
             done
             echo "::set-output name=push::yes"
-          fi
+          fi
```
Comment (dnillovna) [inline .github/workflows/packages.yml:469]:
```suggestion
          fi

```

File: .github/workflows/packages.yml [hunk 10]
```diff
@@ -230,6 +230,67 @@ jobs:
...
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }} 
+        if: steps.check_directory.outputs.files_exists == 'true' && steps.NEdockerdir.outputs.DIR_EMPTY == 'false' && steps.publish.outcome == 'success'
+
+      - name: get docker image version
+        id: docker_name_ver
+        working-directory: packages/${{ matrix.package }}
+        run: |
+          echo "::set-output name=IMAGE_VER::$(jq -r .version package.json)"
+          echo "::set-output name=PACKAGE_NAME::$(jq -r .version package.json | sed -e 's,@datagrok/,,g')"
```
[the remark this answers is not in the document]
Comment (dnillovna) [reply .github/workflows/packages.yml:265]: And, I think, it is better to use lowercased names for the outputs

**Review by dnillovna — CHANGES_REQUESTED** [review]

File: .github/workflows/packages.yml [hunk 11]
```diff
@@ -230,6 +230,67 @@ jobs:
...
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
         if: github.ref == 'refs/heads/master' && always() && matrix.publish == 'true' && (steps.build.outcome == 'success' || steps.build.outcome == 'skipped') && steps.check.outcome == 'success' && steps.install.outcome == 'success'
+
+      - name: Check dockerfiles directory existence
+        id: check_directory
+        uses: andstor/file-existence-action@v2
+        with:
+          files: "dockerfiles"  
+      - name: check if dockerfies directory not empty 
+        id: NEdockerdir
```
[the remark this answers is not in the document]
Comment (ksumag) [reply .github/workflows/packages.yml:240]: NotEmpty docker dir. 
ok, I'll change

**Pull request merged**